test/negotiation/utils/NegotiationTest.js

```diff
@@ -0,0 +1,473 @@
+        before(async function() {
+          clientStream = await ctx.createClientStream();
+
+          await ctx.client.addStream(clientStream);
+          await ctx.erizo.publishStream(clientStream);
+
+          offer = await ctx.client.getSignalingMessage();
+
+          await ctx.erizo.processSignalingMessage(offer);
+          answer = ctx.erizo.getSignalingMessage();
+
+          await ctx.client.processSignalingMessage(answer);
+
+          if (!ctx.candidates) {
+            await ctx.client.waitForCandidates();
+            ctx.candidates = await ctx.client.getAllCandidates();
+            for (const candidate of ctx.candidates) {
+              await ctx.erizo.processSignalingMessage(candidate);
+            }
+          }
+
+          await ctx.client.waitForConnected();
+          await ctx.erizo.waitForReadyMessage();
+          erizoMessages = ctx.erizo.getSignalingMessages();
+        });
+
+        ctx.checkClientSentCorrectOffer(() => offer);
+        ctx.checkErizoSentCorrectAnswer(() => answer);
+        ctx.checkClientSentCandidates();
+        ctx.checkErizoFinishedSuccessfully();
+        ctx.checkClientFSMStateIsStable();
+        ctx.checkClientHasNotFailed();
+      });
+    };
+
+    ctx.unpublishStreamStep = function() {
+      describe('Unpublish One Client Stream', function() {
+        let offer, answer, candidates, erizoMessages, clientStream, label;
+        before(async function() {
+          clientStream = ctx.getFirstClientStream();
+
+          await ctx.client.removeStream(clientStream);
+          await ctx.erizo.unpublishStream(clientStream);
+
+          offer = await ctx.client.getSignalingMessage();
+
+          await ctx.erizo.processSignalingMessage(offer);
+          answer = ctx.erizo.getSignalingMessage();
+
+          await ctx.client.processSignalingMessage(answer);
+
+          if (!ctx.candidates) {
+            await ctx.client.waitForCandidates();
+            ctx.candidates = await ctx.client.getAllCandidates();
+            for (const candidate of ctx.candidates) {
+              await ctx.erizo.processSignalingMessage(candidate);
+            }
+          }
+
+          await ctx.client.waitForConnected();
+          await ctx.erizo.waitForReadyMessage();
+          erizoMessages = ctx.erizo.getSignalingMessages();
+
+         await ctx.deleteClientStream(clientStream);
+        });
+
+        ctx.checkClientSentCorrectOffer(() => offer);
+        ctx.checkErizoSentCorrectAnswer(() => answer);
+        ctx.checkClientSentCandidates();
+        ctx.checkErizoFinishedSuccessfully();
+        ctx.checkClientFSMStateIsStable();
+        ctx.checkClientHasNotFailed();
+      });
+    };
+
+    ctx.subscribeToErizoStreamStep = function() {
+      describe('Subscribe to One Erizo Stream', function() {
+        let offer, answer;
+
+        before(async function() {
+          remoteStream = await ctx.createRemoteStream();
```

Comment:
I think `remoteStream` in this context is a bit confusing. I think maybe `erizoStream` would be more descriptive since it's a stream that originates in Erizo in the context of the negotiation.

Reply:
yes, you're right that it will be more descriptive 👍